webapp/src/pages/Users/Users.tsx

```diff
@@ -1,66 +1,34 @@
-import './Users.css';
+import { IconVariant, RowItem } from '../../shared/components';
 import {
-  IconVariant,
-  NavigationBar,
-  RowItem,
-  RowsList,
-  SearchForm,
-  SmallAvatar,
-} from '../../shared/components';
-import { useState } from 'react';
-import {
-  USER_URL,
   USERS_EP_URL,
   USERS_URL,
   WILDCARD_AVATAR_URL,
 } from '../../shared/urls';
-import { Link } from 'react-router-dom';
-import { instanceOfArrayTyped } from '../../shared/types';
 import { instanceOfUser, User } from '../../shared/generated';
+import { DispatchPage } from '../../shared/components/index';
 
-function instanceOfUserArray(value: object): boolean {
-  return instanceOfArrayTyped(value, instanceOfUser);
-}
-
-const mapUsersToRows = (users: User[]): RowItem[] => {
-  return users.map((user) => {
-    return {
-      iconVariant: IconVariant.ARROW_FORWARD,
-      avatarProps: {
-        url: user.avatarId
-          ? `${USERS_EP_URL}/${user.id}/avatar`
-          : WILDCARD_AVATAR_URL,
-        status: 'offline',
-      },
-      url: `${USERS_URL}/${user.id}`,
-      title: user.username,
-      subtitle: 'level x',
-      key: user.id,
-    };
-  });
+const mapUserToRow = (user: User): RowItem => {
+  return {
+    iconVariant: IconVariant.ARROW_FORWARD,
+    avatarProps: {
+      url: user.avatarId
+        ? `${USERS_EP_URL}/${user.id}/avatar`
+        : WILDCARD_AVATAR_URL,
+      status: 'offline',
+    },
+    url: `${USERS_URL}/${user.id}`,
+    title: user.username,
+    subtitle: 'level x',
+    key: user.id,
+  };
 };
 
 export default function Users() {
-  const [users, setUsers] = useState<User[]>([]);
-
   return (
-    <div className="users">
-      <div className="users-avatar">
-        <Link to={USER_URL}>
-          <SmallAvatar url={`${USERS_EP_URL}/avatar`} />
-        </Link>
-      </div>
-      <div className="users-search">
-        <SearchForm url={`${USERS_EP_URL}?search=`} setChange={setUsers} />
-      </div>
-      <div className="users-rows">
-        {instanceOfUserArray(users) && (
-          <RowsList rows={mapUsersToRows(users)} />
-        )}
-      </div>
-      <div className="users-navigation">
-        <NavigationBar />
-      </div>
-    </div>
+    <DispatchPage
+      dataValidator={instanceOfUser}
+      genericEndpointUrl={USERS_EP_URL}
+      dataMapper={mapUserToRow}
+    />
```

Comment:
question: If I understand correctly, we will use the same layout for the Users, Play and Chat pages?

thought: As long as we keep this in mind, it will allow us to have a consistent UI between these pages and reduce code duplication.

Reply:
That is the general idea, yes. But who knows, we'll see as we go forward.